app/assets/javascripts/components/samples.jsx

```diff
@@ -21,14 +22,33 @@ class Samples extends React.Component {
       sampleIdsParams: this.fetchParams('ids') || [],
       allSamples: [],
       allProjects: [],
-      sort_query: currentSort.sort_query ? currentSort.sort_query  : `sort_by=${this.defaultSortBy}`,
+      sort_by: this.fetchParams('sort_by') || 'created_at,desc',
       pagesLoaded: 0,
       pageEnd: false,
       initialFetchedSamples: [],
       loading: false,
       isRequesting: false,
       displayEmpty: false
     };
+    this.sortCount = 0;
+  }
+
+  sortSamples() {
+    this.sortCount += 1;
+    let new_sort;
+    if(this.sortCount === 3) {
+      this.sortCount = 0;
+      new_sort = 'created_at,desc';
```

Comment:
This means the 3rd time you click on the arrow, the samples are again sorted by date and not name. I think the user would not understand what's happening, given that the arrow is next to "Name". Is that what we want @eggerdesigns ?

Reply:
@cdebourcy when you click on it the third time I think that the arrow should disappear and the list is once again sorted by most recent upload. This isn't super intuitive but it gets the job done.

Reply:
Makes sense. @cyrielo so please hide the arrow the 3rd time